scm/driver/gitlab/webhook.go

```diff
@@ -420,6 +430,61 @@ func parseTimeString(timeString string) time.Time {
 	return t
 }
 
+func convertPipelineHook(src *pipelineHook) *scm.PipelineHook {
+	namespace, name := scm.Split(src.Project.PathWithNamespace)
+	return &scm.PipelineHook{
+		Repo: scm.Repository{
+			ID:        strconv.Itoa(src.Project.ID),
+			Namespace: namespace,
+			Name:      name,
+			Clone:     src.Project.GitHTTPURL,
+			CloneSSH:  src.Project.GitSSHURL,
+			Link:      src.Project.WebURL,
+			Branch:    src.Project.DefaultBranch,
+			Private:   false, // TODO how do we correctly set Private vs Public?
+		},
+		Commit: scm.Commit{
+			Sha:     src.ObjectAttributes.SHA,
+			Message: src.Commit.Message,
+			Author: scm.Signature{
+				Login:  src.User.Username,
+				Name:   src.Commit.Author.Name,
+				Email:  src.Commit.Author.Email,
+				Avatar: src.User.AvatarURL,
+			},
+			Committer: scm.Signature{
+				Login:  src.User.Username,
+				Name:   src.User.Name,
+				Email:  src.User.Email,
+				Avatar: src.User.AvatarURL,
+			},
```

Comment:
can the commit author be different from the user?  I suspect they can be if a pipeline is manually executed

Reply:
updated in latest changes